### ISSUE TITLE
Create a way for `has-one` relationship use `.create` (created after #472)

### ISSUE THREAD

**Issue by FCO** [FCO/Red#475]
take a look at #472

**Comment by FCO** [comment]
> The interesting part of it is that it is actually possible set attribute type to the model it is refers to. I.e.:
> ... 
> Foo.attr.WHICH # MyModel

@vrurg but on that case, Foo.attr should return a Red::Column to make it possible to be used inside .map, .grep, etc...

**Comment by FCO** [comment]
As @jonathanstowe have said on the original issue, I think returning a model type would be the best way, it could use the .join-on (the same method we use to find joins) to create it with the right external ids... I'll give it a try as soon as possible!

If we do that change, should that generate a new API version?

[![](https://api.gh-polls.com/poll/01FKP9ZJH7BJTF8WYT69761YVJ/Yes%2C%20it%20changes%20the%20behaviour%2C%20so%20it%20should%20be%20a%20different%20API%20version)](https://api.gh-polls.com/poll/01FKP9ZJH7BJTF8WYT69761YVJ/Yes%2C%20it%20changes%20the%20behaviour%2C%20so%20it%20should%20be%20a%20different%20API%20version/vote)
[![](https://api.gh-polls.com/poll/01FKP9ZJH7BJTF8WYT69761YVJ/No%2C%20no%20one%20is%20testing%20the%20returned%20type%2C%20only%20if%20it%E2%80%99s%20true%20or%20if%20it%E2%80%99s%20defined)](https://api.gh-polls.com/poll/01FKP9ZJH7BJTF8WYT69761YVJ/No%2C%20no%20one%20is%20testing%20the%20returned%20type%2C%20only%20if%20it%E2%80%99s%20true%20or%20if%20it%E2%80%99s%20defined/vote)
[![](https://api.gh-polls.com/poll/01FKP9ZJH7BJTF8WYT69761YVJ/No%2C%20no%20one%20is%20using%20Red)](https://api.gh-polls.com/poll/01FKP9ZJH7BJTF8WYT69761YVJ/No%2C%20no%20one%20is%20using%20Red/vote)
[![](https://api.gh-polls.com/poll/01FKP9ZJH7BJTF8WYT69761YVJ/I%20don%E2%80%99t%20know%20)](https://api.gh-polls.com/poll/01FKP9ZJH7BJTF8WYT69761YVJ/I%20don%E2%80%99t%20know%20/vote)

**Comment by jonathanstowe** [comment]
I don't think it needs a different API version, because Empty is false, a type object is false.

**Comment by FCO** [comment]
I'm still trying to figure out why... but my change is working if I define the referencing's model as type and does not work if I define it as string:

<img width="1230" alt="image" src="https://user-images.githubusercontent.com/99834/140614600-ab7ad38b-f292-4ad1-af62-cbbd8dcab915.png">

**Comment by FCO** [comment]
for now: https://github.com/FCO/Red/pull/523

**Comment by FCO** [comment]
There are still some bugs on it... but it seems to be mostly working...
